### PR TITLE
#2834 missing KUBECONFIG file does not cause CLI to break anymore

### LINKED ISSUE #2834: Keptn CLI (master branch) fails if wrong KUBECONFIG is set
I have set my KUBECONFIG to a file that does not exist (yet), and keptn cli.

To reproduce:
* Set KUBECONFIG, e.g., `export KUBECONFIG=/a/b/c.yml`
* Build Keptn CLI from master branch
* Run `keptn version`

Output:
```
$ export KUBECONFIG=/a/b/c.yml
$ keptn version
2020/12/18 14:35:54 Cannot find file /a/b/c.yml
```

Expected behaviour: CLI should work, and not worry about KUBECONFIG (it should treat it like there is no kubernetes cluster available)

Workaround for now: make sure KUBECONFIG is properly set

### PR DESCRIPTION
**Pull request by bacherfl**
Closes #2834 
Signed-off-by: Florian Bacher <florian.bacher@dynatrace.com>

### CODE REVIEW

File: cli/pkg/credentialmanager/handler.go [hunk 1]
```diff
@@ -171,12 +171,14 @@ func getCurrentContextFromKubeConfig() error {
 
 	fileContent, err := file.ReadFile(kubeconfig)
 	if err != nil {
-		return err
+		logging.PrintLog("Warning: could not open KUBECONFIG file: "+err.Error(), logging.InfoLevel)
+		return nil
```
Comment (christian-kreuzberger-dtx) [inline cli/pkg/credentialmanager/handler.go:175]: I guess by returning `nil` here, 
```
		err := getCurrentContextFromKubeConfig()
		if err != nil {
			log.Fatal(err)
		}
		checkForContextChange(cliConfigManager, autoApplyNewContext)
```
won't exit, and therefore Issue #2834 will be solved?

File: cli/pkg/credentialmanager/handler.go [hunk 2]
```diff
@@ -171,12 +171,14 @@ func getCurrentContextFromKubeConfig() error {
 
 	fileContent, err := file.ReadFile(kubeconfig)
 	if err != nil {
-		return err
+		logging.PrintLog("Warning: could not open KUBECONFIG file: "+err.Error(), logging.InfoLevel)
```
Comment (christian-kreuzberger-dtx) [inline cli/pkg/credentialmanager/handler.go:174]: I'm not sure if I was doing it wrong, but I can't get this log message to be displayed. Should this be shown if I do the following
```
export KUBECONFIG=/a/b/c/d.yaml
./keptn version -v
```
?

File: cli/pkg/credentialmanager/handler.go (hunk 1 above)
Comment (bacherfl) [reply cli/pkg/credentialmanager/handler.go:175]: exactly. in this case, the current kubernetes context will be set to `""`, and the CLI will proceed to use the currently active credentials

File: cli/pkg/credentialmanager/handler.go (hunk 2 above)
Comment (bacherfl) [reply cli/pkg/credentialmanager/handler.go]: Just checked it and you are right - that message is not displayed. I have just committed a change to display this message via `stderr`

File: cli/pkg/credentialmanager/handler.go [hunk 3]
```diff
@@ -171,12 +170,14 @@ func getCurrentContextFromKubeConfig() error {
 
 	fileContent, err := file.ReadFile(kubeconfig)
 	if err != nil {
-		return err
+		fmt.Fprintf(os.Stderr, "Warning: could not open KUBECONFIG file: "+err.Error())
```
Comment (christian-kreuzberger-dtx) [inline cli/pkg/credentialmanager/handler.go:173]: works now! But still needs a `\n` / newline at the end

File: cli/pkg/credentialmanager/handler.go [hunk 4]
```diff
@@ -171,12 +170,14 @@ func getCurrentContextFromKubeConfig() error {
...
+		return nil
 	}
 
 	err = yaml.Unmarshal([]byte(fileContent), &kubeConfigFile)
 	if err != nil {
-		log.Fatalf("Unmarshal: %v", err)
+		fmt.Fprintf(os.Stderr, "Warning: could not parse KUBECONFIG file: "+err.Error())
```
Comment (christian-kreuzberger-dtx) [inline cli/pkg/credentialmanager/handler.go:179]: Same here, we'll need a `\n` or newline at the end

**Review by christian-kreuzberger-dtx — CHANGES_REQUESTED** [review]
Two more comments, after that this can be merged :+1:

**Review by christian-kreuzberger-dtx — APPROVED** [review]
LGTM (please squash merge)

**Pull request merged**